internal/cmdrender/executor.go

```diff
@@ -77,24 +84,39 @@ func (e *Executor) Execute(ctx context.Context) error {
 		return err
 	}
 
-	pkgWriter := &kio.LocalPackageReadWriter{PackagePath: string(root.pkg.UniquePath)}
-
 	// format resources before writing
 	_, err = filters.FormatFilter{UseSchema: true}.Filter(resources)
 	if err != nil {
 		return err
 	}
 
-	err = pkgWriter.Write(resources)
-	if err != nil {
-		return fmt.Errorf("failed to save resources: %w", err)
+	if e.Output == nil {
+		pkgWriter := &kio.LocalPackageReadWriter{PackagePath: string(root.pkg.UniquePath)}
+		err = pkgWriter.Write(resources)
+		if err != nil {
+			return fmt.Errorf("failed to save resources: %w", err)
+		}
+	} else {
```

Comment:
Not a big deal, but I think we can unify (in-place directory) writing with `out-dir`.
Also, writing of resources is now split (some here in this function) and some outside it. Need to think more before suggesting a good way to do it.

Reply:
As discussed offline, we will address this later.

Reply:
I thought about your proposal where we can write the resources in wrapped format in `e.Output` for all the cases `in-place` as well as `out-of-place` and pass it to the next layer (cmdRunner), but realized for in-place pruning etc, we will have to lift the `hctx` as well because input/output files are tracked in `hctx`.

Another idea: why not move the call `cmdutil.WriteFnOutput(..)` from `cmdRunner` to `Executor` here itself. Then code being spread in two location will be solved. Only downside is wiring `dest` at the executor level from runner which is a small cost to pay but acceptable.